### DISCUSSION TITLE
Any example of running native Swift with Tauri V2?

### DISCUSSION THREAD

**Discussion by gergomiklos** [tauri-apps/tauri#9581] [Q&A]
I want to capture and stream live system audio on MacOS, and the only possible solution for this is to use the native ScreenCaptureKit with Swift. There are two possible approaches: one is to start a Swift websocket server and send audio data there, and the other is to send the audio data directly to the JS frontend and handle it there. Is there a Swift plugin example similar to these? Or any Swift plugin example?

**Comment by sam-brainstream** [comment]
ever figure this out?

**Comment by offbyseven** [comment]
I don’t think there are any Swift-plugins for this unfortunately that I know of. There are a few other options though that might work:

Since your backend is already Rust, you can use the [screencapturekit-rs](https://github.com/svtlabs/screencapturekit-rs) crate, which wraps ScreenCaptureKit and can capture screen content and the audio that goes with it. 

If you specifically need Swift-side APIs, you can use the[ swift-rs](https://github.com/Brendonovich/swift-rs) crate instead. You write the capture in Swift, expose a C ABI, and call it from Tauri.

You can also use a desktop recording SDK to record audio and video directly from a MacOS device. This can deliver audio either through a websocket or direct-to-app callback like you were suggesting, and will be the easiest in terms of set-up/maintenance.
